zingolib/src/wallet/transaction_record.rs

```diff
@@ -25,47 +25,67 @@ use crate::{
     },
 };
 
-///  Everything (SOMETHING) about a transaction
+/// This Capabilities perspective on a Transaction
+/// Key:   [ChainPublic|Outgoing|Incoming]
+///   ChainPublic: this data can be obtained directly from the chain, it has no privacy properties
+///   Outgoing:    this data can be Viewed by the Capability that createed this Transaction
+///   Incoming:    this data can be Viewed by the Recipient Capability that is encumbered by the note
+///  NOTE: A received note on an outgoing transaction was formerly called "change".
 #[derive(Debug)]
 pub struct TransactionRecord {
     /// the relationship of the transaction to the blockchain. can be either Broadcast (to mempool}, or Confirmed.
+    ///  # ChainPublic
     pub status: zingo_status::confirmation_status::ConfirmationStatus,
 
     /// Timestamp of Tx. Added in v4
+    ///  # ChainPublic
     pub datetime: u64,
 
     /// Txid of this transaction. It's duplicated here (It is also the Key in the HashMap that points to this
-    /// WalletTx in LightWallet::txs)
+    /// TransactionRecord in LightWallet::TransactionRecordsById)
+    ///  # ChainPublic
     pub txid: TxId,
 
-    /// List of all nullifiers spent by this wallet in this Tx.
-    pub spent_sapling_nullifiers: Vec<sapling_crypto::Nullifier>,
+    /// List of all Sapling nullifiers used by this Capability to add value to the Sapling Value Pool.
+    ///  # Outgoing
```

Comment:
lets have a discussion about this PR, i have left some comments in chat. Outgoing here highlights the type of confusion we might unleash by defining inputs as nullifiers, (and only nullifiers). nullifiers are actually public on chain so should fall into the category of public. but i suppose there is confusion here that the value comes from a note/output only seen by the "Outgoing" capability. however, a nullifier is public and a link to that output, not the output itself or the value it holds

Reply:
i like the clarity that these nullifiers are inputs to THIS transaction
 
 im confused about what ///  # Outgoing means.